File: fizzbuzz_1.rb
class FizzBuzzOne
  attr_accessor :result, :limit
  private :result, :limit

  def initialize(limit)
    @limit = limit
    @result = ""
  end

  def display_series
    (1..limit).each { |x| @result += "#{define_number(x)} " }
    puts result
    result
  end

  private

  def define_number(x)
    number = ""
    number += "fizz" if (x % 3).zero?
    number += "buzz" if (x % 5).zero?
    number != "" ?  number : x.to_s
  end
end

FizzBuzzOne.new(100).display_series

File: fizzbuzz_1.js
var FizzBuzzOne = (function() {
  function FizzBuzzOne(limit) {
    this.limit = limit;
    this.result = "";
  };

  FizzBuzzOne.prototype.displaySeries = function() {
    for (var i = 1; i <= this.limit; i++) {
      this.result += this.defineOutput(i) + " ";
    }
    // console.log(this.result);
  };

  FizzBuzzOne.prototype.defineOutput = function(x) {
    var number = "";
    if (x % 3 === 0) number += "fizz";
    if (x % 5 === 0) number += "buzz";

    return number === "" ? x : number;
  };

  return FizzBuzzOne;
})();

fizzbuzz = new FizzBuzzOne(10000000);
fizzbuzz.displaySeries();
